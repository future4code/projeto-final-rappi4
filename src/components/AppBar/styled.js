import styled from 'styled-components'

export const AppBarStyled = styled.div`
    height: 44px;
    width: 50vh;
    margin-bottom: 1px;
    display: grid;
    grid-template-columns: 1fr 2fr 1fr;
    justify-content: center;
    align-items: center;
    border-bottom: 1px solid rgba(0, 0, 0, 0.5);
`

export const TitleAppBarStyled = styled.div`
    font-size: 16px;
    font-weight: normal;
    font-stretch: normal;
    font-style: normal;
    line-height: normal;
    letter-spacing: -0.39px;
    text-align: center;
    color: var(--black);
`

export const ArrowBackStyled = styled.div`
    justify-content:center;
    align-items:center;
    align-self:center;
    text-align: center;
`