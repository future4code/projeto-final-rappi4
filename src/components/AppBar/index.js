import React from 'react'
import { AppBarStyled, TitleAppBarStyled, ArrowBackStyled } from './styled'

export default function AppBar(props) {
    return (
        <div>
            <AppBarStyled>
                {/* PASSAR COMO PROPS O DISPATCH GO BACK */}
                <ArrowBackStyled onClick={props.onClickBackIcon}>
                    {props.ArrowBackVisible}
                </ArrowBackStyled>
                {/* PASSAR COMO PROPS O TITULO DA PAGINA */}
                <TitleAppBarStyled>{props.title}</TitleAppBarStyled>
                <div></div>
            </AppBarStyled>
        </div>
    )
}
{/* <AppBar
    title={'Meu Carrinho'}
    onClickBackIcon={this.clickBackIcon}
    ArrowBackVisible={<ArrowBack />}
/>
import ArrowBack from '@material-ui/icons/ArrowBackIos';
 */}