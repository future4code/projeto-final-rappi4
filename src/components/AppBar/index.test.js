import React from 'react';
import renderer from 'react-test-renderer';
import AppBar from './';

describe("AppBar", () => {
    it("should render ", () => {

        const mockProps = {
            onClickBackIcon: "tem",
            title: "tem tb",
            ArrowBackVisible: "olha sรณ, temos",
        }

        const tree = renderer.create(<AppBar
            onClickBackIcon={mockProps.onClickBackIcon}
            title={mockProps.title}
            ArrowBackVisible={mockProps.ArrowBackVisible}
            />).toJSON();

        expect(tree).toMatchSnapshot();
    })
})