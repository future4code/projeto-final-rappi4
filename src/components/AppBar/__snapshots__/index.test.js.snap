// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBar should render  1`] = `
<div>
  <div
    className="sc-bdVaJa fdybHj"
  >
    <div
      className="sc-htpNat ixxngO"
      onClick="tem"
    >
      olha sรณ, temos
    </div>
    <div
      className="sc-bwzfXH ebWYhD"
    >
      tem tb
    </div>
    <div />
  </div>
</div>
`;
